source/common/http/http2/codec_impl.cc

```diff
@@ -456,6 +500,8 @@ int ConnectionImpl::onStreamClose(int32_t stream_id, uint32_t error_code) {
     }
 
     connection_.dispatcher().deferredDelete(stream->removeFromList(active_streams_));
+    nghttp2_session_consume(session_, stream_id, stream->unconsumed_bytes_);
```

Comment:
OK thanks for looking into this. I might add a comment here for the next person that comes along and wonders about this.